src/coord/src/catalog/migrate.rs

```diff
@@ -50,16 +52,32 @@ where
     Ok(())
 }
 
+lazy_static! {
+    static ref VER_0_9_1: Version = Version::new(0, 9, 1);
+}
+
 pub(crate) fn migrate(catalog: &mut Catalog) -> Result<(), anyhow::Error> {
     let mut storage = catalog.storage();
+    let catalog_version = storage.get_catalog_content_version()?;
+    let catalog_version = match Version::parse(&catalog_version) {
+        Ok(v) => v,
+        // Catalog content versions changed to semver after 0.8.3, so all
+        // non-semver versions are less than that.
+        Err(_) => Version::new(0, 0, 0),
+    };
+
     let mut tx = storage.transaction()?;
     // First, do basic AST -> AST transformations.
     rewrite_items(&tx, |stmt| {
         ast_rewrite_type_references_0_6_1(stmt)?;
         ast_use_pg_catalog_0_7_1(stmt)?;
         ast_insert_default_confluent_wire_format_0_7_1(stmt)?;
+        if catalog_version < *VER_0_9_1 {
```

Comment:
@benesch Added a version check to the migration to ensure the migrations remain idempotent.